### ISSUE TITLE
Create preprocessing function

### ISSUE THREAD

**Issue by MBattagl** [MethodsConsultants/tibbletest#11]
Could include the following features:
- `janitor::clean_names()`
- Label cleaning. At the very least collapsing labels with different cases. Would be nice to automatically reformat all labels to some nice format for tables, but not sure how realistic this is.
- Convert 0/1 and yes/no to logical with `convert_logical()`, addresses #5 

Should be very noisy and message everything that it changes

**Comment by MBattagl** [comment]
@scheidec I think initializing variable labels would be done here. Have the user pass in a named list with variable/name, with a default of `snakecase::to_title_case` for all unspecified variables?

**Comment by scheidec** [comment]
Agreed it should be a part of this function.  I've used `sjlabelled::set_label` in the past to add label metadata to variable names: https://rdrr.io/cran/sjlabelled/man/set_label.html.  Do you know of any alternatives in the tidyverse?

**Comment by MBattagl** [comment]
No, I was also thinking about `sjlabelled`. I think that is likely our best option.

**Comment by scheidec** [comment]
So I'm thinking the syntax for this could be:

```
preprocess(df, labels = NULL, clean_names = TRUE, convert_logical = TRUE)
```

Any other functionality or arguments worth adding here?

**Comment by MBattagl** [comment]
We were also thinking of adding some kind of value cleaning if the only difference is the case (Yes/yes/no), or other simple cases. This is more complex than the other tasks, so if we try to do this it could be in its own MR later

**Comment by scheidec** [comment]
@MBattagl any preference for how to present the messaging in the output?  Here's an example of what I have so far, but I'm not quite happy with it yet:

![Screen Shot 2019-06-26 at 12 07 55 PM](https://user-images.githubusercontent.com/19337893/60196449-31424680-980b-11e9-9bf4-048154b9d813.png)

We could change the coloring of the messages to separate the output for each step that is performed?

**Comment by MBattagl** [comment]
Looking good, the tables with changes are awesome. When we update the example data we should make the names uglier to show off the feature. 

I like the idea of color coating the output messages. We could also try more clearly defined sections, with a blank row in between each step. Maybe even a header row with the name of the step?

**Comment by MBattagl** [comment]
If the user wants to change one of those default labels, what should the syntax look like? Would using `...`, `enquos(...)`, and `!!!` allow for a `rename` type syntax? The other option would be passing in a named list?

In either case, would the user need to pass in the original name or the cleaned name to identify the column? Could we make it so either is acceptable?

**Comment by scheidec** [comment]
I'm setting it up so the user passes a named list for all variables/labels in this function.  The tricky thing is if they specify only a couple or miss one variable, what happens? should all the rest use the default title case version of the variable name as the label?  I'd prefer the user pass the variable names for everything in the downstream functions, and the labels will only used for presentation in the tables.

**Comment by MBattagl** [comment]
My idea for how I'd like to use it in practice would be to first run the function with defaults, and look at what labels were generated by default. 90% of the default labels should be good as they are, but I'll want to change a few. Then I'll run the function again passing in only the labels that I want to override.

```
example_dat %>% preprocess()
example_dat <- example_dat %>% preprocess(labels = list(treat2 = "Alternative Treatment"))
```

**Comment by MBattagl** [comment]
My other question is should these be equivalent, or should only one be allowed?

```
example_dat %>% preprocess(labels = list("sugar   factor" = "Blood Sugar"))
example_dat %>% preprocess(labels = list(sugar_factor = "Blood Sugar"))
```

**Comment by scheidec** [comment]
The functionality for overriding defaults should work as you specified.  I'd prefer only the original variable name be allowed, not the cleaned variable name, since the variable name cleaning is happening at runtime at the same time as the variable labeling.

**Comment by MBattagl** [comment]
It would be nice to allow the cleaned names because users would be looking at the final message output table when specifying the labels to override. 

Would this work to make the two equivalent?
```
names(labels) <- labels %>%
  names() %>%
  janitor::make_clean_names()
```

**Comment by MBattagl** [comment]
@scheidec What are your thoughts on either removing completely, or moving the variable labeling to the `format_tbl()` function? It doesn't seem like it would be easy to retain these variable labels through the `descriptives()` function.

**Comment by scheidec** [comment]
I'd be okay with moving it to the `format_tbl()` function, and it might make more sense there so we don't have to think about labels until after the initial table is created.  Especially if it makes it easier to work with in the `descriptives()` function.
